erpnext/stock/doctype/warehouse/warehouse.js

```diff
@@ -37,7 +37,7 @@ frappe.ui.form.on("Warehouse", {
 				function() { convert_to_group_or_ledger(frm); }, 'fa fa-retweet', 'btn-default')
 		}
 		
-		frm.toggle_enable(['is_group', 'company'], false);
+		frm.toggle_enable(['is_group'], false);
```

Comment:
@nmimsnikhil  add is new check otherwise it will create inconsistency in transactions.
```suggestion
		frm.toggle_enable(['is_group'], false);
		if (frm.is_new()) {
			frm.toggle_enable(['company'], true);
		}
```